internal/services/synapse/validate/artifacts.go

```diff
@@ -0,0 +1,24 @@
+package validate
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+func SynapsePipelineAndTriggerName() pluginsdk.SchemaValidateFunc {
+	return func(i interface{}, k string) (warnings []string, errors []error) {
+		value, ok := i.(string)
+		if !ok {
+			errors = append(errors, fmt.Errorf("expected %q to be a string", k))
+			return
+		}
+
+		if !regexp.MustCompile(`^[A-Za-z0-9_][^<>*#.%&:\\+?/]*$`).MatchString(value) {
+			errors = append(errors, fmt.Errorf("invalid name, see https://docs.microsoft.com/en-us/azure/data-factory/naming-rules %q: %q", k, value))
```

Comment:
this should go the region agnostic page:

```suggestion
			errors = append(errors, fmt.Errorf("invalid name, see https://docs.microsoft.com/azure/data-factory/naming-rules %q: %q", k, value))
```

Reply:
Wow. Good findings.